src/solver/attract.jl

```diff
@@ -0,0 +1,309 @@
+# using Test
+
+"""
+    detect_doubleangle1(arg, var)
+
+Detects if ``2sin(x)cos(x)`` is present in any form in an expression
+However, this function is currently useless since we can just 
+apply the rewrite rule to the expression and see if it works or not.
+For future reference, see if there are any positives of detecting the presence
+of the identity first in the expression. Returns true if found,
+false if not.
+
+# Arguments
+- arg: Should be SymbolicUtils.BasicSymbolic{Real} only.
+- var: Num and should satisfy is_singleton
+
+# Examples
+```jldoctest
+julia> detect_doubleangle1(unwrap(2sin(x)cos(x)), x)
+true
+```
+"""
+function detect_doubleangle1(arg, var)
+    !iscall(arg) && return false
+    oper_arg = operation(arg)
+
+    found = [false, false, false]
+
+    c = 1
+    sub_args = arguments(arg)
+    oper_arg != (*) && return false
+    length(sub_args) != 3 && return false
+
+    sin_content = NaN
+    cos_content = NaN
```

Comment:
that is an interesting default :`)

Reply:
i think this function is unused anyways, we bruteforce trig identities using the ones i found in simplify